pragma-node/src/utils/ws.rs

```diff
@@ -173,18 +274,11 @@ where
                             return Err(e);
                         }
                     }
-                },
-                // Messages from the server to the client
-                maybe_server_msg = self.notify_receiver.recv() => {
-                    if let Some(server_msg) = maybe_server_msg {
-                        let _ = self.sender.send(server_msg).await;
-                    }
-                },
-                // Exit signal
+                }
+
+                // Check if the channel has been closed
                 _ = self.exit.1.changed() => {
                     if *self.exit.1.borrow() {
-                        self.sender.close().await.ok();
-                        self.closed = true;
```

Comment:
why remove that ?

Reply:
*  `sender` is no longer accessible through `self`, now we handle only an mspc channel that redirects messages in & out to the socket (avoid bottlenecks while processing a client)
* so we no longer need to close the sender here, it is actually done when we drop the subscriber after sending the `close` notification on the watch channel
* `closed` boolean was actually useless, not sure why we added it so removed